### ISSUE TITLE
Protobuf shared library

### ISSUE THREAD

**Issue by havardstien-FFI** [mesonbuild/wrapdb#837]
As far as I can tell, it is not currently possible to build the protobuf wrap as a shared library. I specifically need protobuf as shared library, and I am not able to use meson as my build system because of it.

**Comment by neheb** [comment]
It it possible with cmake?

**Comment by havardstien-FFI** [comment]
> It it possible with cmake?

I don't know, but vcpkg can create shared libraries of protobuf.

**Comment by jpakkane** [comment]
What happens if you do try to build it as a shared lib? Is it just a question of missing symbol exports or something else?

**Comment by havardstien-FFI** [comment]
I have looked into it further. I am building on Windows, and it seems protobuf forces static build on Windows. 
Line 5 in protobuf's meson.build:
`ttype = cc.get_argument_syntax() == 'msvc' ? 'static_library' : 'library'`

**Comment by jpakkane** [comment]
That is probably due to the fact that Protobuf requires some magic compiler defines to make it work with MSVC but nobody has yet dove in to reverse engineer what those need to be (it might even be possible that you can't build the library shared without code changes, though I don't know if that is the case here).

**Comment by neheb** [comment]
@havardstien-FFI still interested?

**Comment by havardstien-FFI** [comment]
@neheb yes. I have thought about trying to make it work myself, but I don't think I will have time to attempt it.
